### ISSUE TITLE
CYGWIN - docker_image hook mount point incorrect

### ISSUE THREAD

**Issue by mdeweerd** [pre-commit/pre-commit#2938]
### search you tried in the issue tracker

is:issue docker_image  cygwin  

### describe your issue

I am using pre-commit under cygwin which generally works file.

However for a docker file it is indicated that the mount point is '-v$PWD:/src'.  I think that $PWD it the path as cygwin reports it, but it is window's docke r that is run, so it should be the windows path which can be found with `(cygpath -w $pwd)`.

As the path is wrong the 'shellscript' hook shown below reports: "protected/scripts/loadLT4.sh: protected/scripts/loadLT4.sh: openBinaryFile: does not exist (No such file or directory)"

The following is how I made the corresponding hook that relies on Dockerfile work.
I mounted the working directory to "/mnt' and chose that  as the workdir (I could not append a mount point to /src because docker does not accept duplicate mount points).

```yaml
  - repo: https://github.com/koalaman/shellcheck-precommit
    rev: v0.9.0
    hooks:
      - id: shellcheck
        entry: -v.:/mnt --workdir=/mnt docker.io/koalaman/shellcheck:v0.9.0
```

### pre-commit --version

pre-commit 3.3.3

### .pre-commit-config.yaml

```yaml
---
files: (?x)^(( protected|themes)/.*\.(php|py|pl|pm|json|md|sh|yaml|cfg|txt)|[^/]*
  )$
repos:
  - repo: https://github.com/koalaman/shellcheck-precommit
    rev: v0.9.0
    hooks:
      - id: shellcheck
```

**Comment by asottile** [comment]
mixing windows and cygwin binaries is not supported

**Comment by mdeweerd** [comment]
In think that instead of using "-v$PWD:/src" one could simply use "-v.:src" which would fix this and be platform independant.

**Comment by asottile** [comment]
it's not `$PWD` but `os.getcwd()` https://github.com/pre-commit/pre-commit/blob/2bffc0ad85c152a721bc5fe2ba7aec7fbf1bedb5/pre_commit/languages/docker.py#L119

and mounts must be absolute paths so `.` won't work

**Comment by asottile** [comment]
plus you'd be better to use https://github.com/shellcheck-py/shellcheck-py anyway

**Comment by mdeweerd** [comment]
> mounts must be absolute paths so . won't work

Maybe it has been, or it's selective by platform.  AFAIK relative paths work as the source path in --volume or --mount in docker.  I have used "-v .:/some/path' several time and I have seen it used.  The target path has to be absolute.  I could not find a definitive statement on the docker site - the documentation is not telling.

My workaround used a relative path and it works.

Thanks for the pointer to [shellcheck-py](https://github.com/shellcheck-py/shellcheck-py) which is a solution for the case I reported, but I tested pre-commit hooks using docker in the past without success - I did not dig into the issue then, but it was likely the same.
